http-server/src/Main.gren

```diff
@@ -0,0 +1,101 @@
+module Main exposing (main)
+
+import Node
+import Bytes exposing (Bytes)
+import Bytes.Encode as BE
+import Node.Program as Program exposing (Program)
+import Node.Http as Http
+
+
+main : Program Model Msg
+main =
+    Program.define
+        { init = init
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Model =
+    { stdout : Stream
+    , stderr : Stream
+    , server : Maybe Http.Server
+    }
+
+
+type Msg
+    = CreateServerResult (Result Http.CreateServerError Http.Server)
+    | HandleRequest Http.Request
+    {-
+        type alias Request =
+            { headers : Dict String String
+            , body : Bytes
+            , url : Url
+            , method : Method
+            }
+        
+        type Method
+            = GET
+            | HEAD
+            | POST
+            | PUT
+            | DELETE
+            | CONNECT
+            | TRACE
+            | PATCH
+    -}
+
+init : Program.AppInitTask { model : Model, command : Cmd Msg }
+init =
+    Program.await Node.initialize <| \nodeConfig ->
+        Program.await Http.initialize <| \serverPermission ->
+            Program.startProgram
+                { model =
+                    { stdout = nodeConfig.stdout
+                    , stderr = nodeConfig.stderr
+                    , server = Nothing
+                    }
+                , command =
+                    Http.createServer serverPermission
+                        { host = "localhost"
+                        , port = "8765"
+                        }
+                        |> Task.attempt CreateServerResult
```

Comment:
I'm getting into the implementation and since `Http` will already be an effect module, `createServer` could return a `Cmd msg` instead of a task. Might be a nicer API? Not sure if there's significant trade-offs though.

Reply:
I think in this case it makes sense to stay with tasks:

* it can fail
* it is expected to return almost immediately after calling